File: pc-agent/Models/ApiModels.cs
namespace ControlCom.Agent.Models;

public sealed class HealthResponse
{
    public bool Ok { get; init; } = true;
    public string Service { get; init; } = "controlcom-agent";
    public string Version { get; init; } = "1.0.0";
}

public sealed class OkResponse
{
    public bool Ok { get; init; } = true;
}

public sealed class PairRequest
{
    public string Code { get; init; } = string.Empty;
}

public sealed class PairResponse
{
    public string Token { get; init; } = string.Empty;
}

public sealed class PairingInfoResponse
{
    public bool Paired { get; init; }
    public string? Ip { get; init; }
    public int Port { get; init; }
    public string? Code { get; init; }
    public string? QrPayload { get; init; }
}

public sealed class ErrorResponse
{
    public string Error { get; init; } = string.Empty;
    public string? Message { get; init; }
}

public sealed class MuteResponse
{
    public bool Muted { get; init; }
}

public sealed class DisplayModeRequest
{
    public string Mode { get; init; } = string.Empty;
}

public sealed class DisplayModeResponse
{
    public string Mode { get; init; } = string.Empty;
}

public sealed class ShutdownResponse
{
    public bool Ok { get; init; } = true;
    public string Message { get; init; } = "shutdown_scheduled";
    public int WindowsNotified { get; init; }
    public int DelaySeconds { get; init; }
}

File: android/app/src/main/java/com/controlcom/app/data/ApiModels.kt
package com.controlcom.app.data

data class HealthResponse(
    val ok: Boolean,
    val service: String,
    val version: String
)

data class OkResponse(
    val ok: Boolean
)

data class PairRequest(
    val code: String
)

data class PairResponse(
    val token: String
)

data class MuteResponse(
    val muted: Boolean
)

data class DisplayModeRequest(
    val mode: String
)

data class DisplayModeResponse(
    val mode: String
)

data class ShutdownResponse(
    val ok: Boolean,
    val message: String,
    val windowsNotified: Int,
    val delaySeconds: Int
)

data class PairingInfoResponse(
    val paired: Boolean,
    val ip: String?,
    val port: Int,
    val code: String?,
    val qrPayload: String?
)

data class ErrorResponse(
    val error: String,
    val message: String? = null
)
